gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,336 @@
+    
+    Filter MT to bi-allelic SNVs at sites with popmax allele frequency greater than `min_popmax_af`, densify, and write MT.
+
+    :param mt: Raw MatrixTable to use for the subpop analysis
+    :param min_popmax_af: Minimum population max variant allele frequency to retain variant for the subpop QC MatrixTable
+    :return: MatrixTable filtered to variants for subpop analysis
+    """
+    release_ht = hl.read_table(release_sites().path)
+
+    # Filter to biallelic SNVs not in low-confidence regions and with a popmax above min_popmax_af
+    qc_sites = release_ht.filter(
+        (release_ht.popmax.AF > min_popmax_af)
+        & ~release_ht.was_split
+        & hl.is_snp(release_ht.alleles[0], release_ht.alleles[1])
+        & hl.is_missing(lcr_intervals.ht()[release_ht.locus])
+    ).key_by("locus")
+
+    # Densify the MT and include only sites defined in the qc_sites HT
+    mt = mt.select_entries(
+        "END", GT=mt.LGT, adj=get_adj_expr(mt.LGT, mt.GQ, mt.DP, mt.LAD)
+    )
+    mt = densify_sites(mt, qc_sites, hl.read_table(last_END_position.path))
+
+    return mt
+
+
+def filter_subpop_qc(
+    mt: hl.MatrixTable,
+    pop: str,
+    min_af: float = 0.001,
+    min_inbreeding_coeff_threshold: float = -0.25,
+    min_hardy_weinberg_threshold: float = 1e-8,
+    ld_r2: float = 0.1,
+) -> hl.MatrixTable:
+    """
+    Generate the QC MT per specified population.
+    
+    .. note::
+
+        Hard filtered samples are removed before running `get_qc_mt`
+
+    :param mt: The QC MT output by the 'compute_subpop_qc_mt' function
+    :param pop: Population to which the QC MT should be filtered
+    :param min_af: Minimum population variant allele frequency to retain variant in QC MT
+    :param min_inbreeding_coeff_threshold: Minimum site inbreeding coefficient to retain variant in QC MT
+    :param min_hardy_weinberg_threshold: Minimum site HW test p-value to keep
+    :param ld_r2: Minimum r2 to keep when LD-pruning (set to `None` for no LD pruning)
+    :return: Filtered QC MT with sample metadata for the specified population to use for subpop analysis
+    """
+    meta_ht = meta.ht()
+
+    # Add info to the MT
+    info_ht = get_info(split=False).ht()
+    info_ht = info_ht.annotate(
+        info=info_ht.info.select(
+            # No need for AS_annotations since it's bi-allelic sites only
+            **{x: info_ht.info[x] for x in info_ht.info if not x.startswith("AS_")}
+        )
+    )
+    mt = mt.annotate_rows(info=info_ht[mt.row_key].info)
+
+    # Add sample metadata to the QC MT
+    mt = mt.annotate_cols(**meta_ht[mt.col_key])
+
+    # Filter the  QC MT to the specified pop
+    pop_mt = mt.filter_cols(mt.population_inference.pop == pop)
+
+    # Remove hard filtered samples
+    pop_mt = pop_mt.filter_cols(~pop_mt.sample_filters.hard_filtered)
+    pop_mt = pop_mt.filter_rows(hl.agg.any(pop_mt.GT.is_non_ref()))
+    pop_mt = pop_mt.checkpoint(
+                get_checkpoint_path("pop_mt", mt=True))
```

Comment:
I know that you plan to also put in an optional checkpoint path to `get_qc_mt` instead of this persist https://github.com/broadinstitute/gnomad_methods/blob/bbb4098b9f67ee35cf5709d8a848102c57048c4d/gnomad/sample_qc/pipeline.py#L182 but doesn't hurt to add one here too, good call. Also just need to run Black again